pages/Livesite/settings.js

```diff
@@ -1,23 +1,152 @@
-import { LIVESITE_NAVBAR } from '../../constants';
+import Card, {
+  CardHeader,
+  CardButtonContainer,
+  CardTitle,
+  CardContent,
+  CardContentButtonContainer,
+  CancelButton,
+} from '../../components/card';
+import { UploadContainer } from '../../components/modal';
+import Button from '../../components/button';
+import {
+  formatDate,
+  getTimestamp,
+  getHackathons,
+  getLivesiteData,
+  updateLivesiteData,
+  uploadLivesiteLogoToStorage,
+} from '../../utility/firebase';
+import { useAuth } from '../../utility/auth';
+import { LIVESITE_NAVBAR, EDIT } from '../../constants';
 
-export default ({ hackathons }) => (
-  <Page
-    currentPath="Livesite"
-    hackathons={hackathons}
-    navbarItems={LIVESITE_NAVBAR}
-  >
-    <img
-      src="https://i.imgur.com/lEjmB7U.gif"
-      width="480"
-      height="362"
-      alt="panik"
-    />
-    <p>Settings - This area under construction.</p>
-  </Page>
-);
+const Label = styled.p`
+  font-weight: bold;
+`;
+
+export default ({ hackathons }) => {
+  const inputFile = React.createRef();
+  const [livesiteData, setLivesiteData] = useState({});
+  const [isEditing, setisEditing] = useState(false);
+  const [imgFile, setImgFile] = useState();
+  const [fileUpload] = useState({});
+  const { email: user } = useAuth().user;
+
+  // MODAL SUBMIT BUTTON CLICKED (NEW + EDIT)
+  const handleSave = async () => {
+    // 1. uploads to firebase
+    // newobj.imgURL is a file object
+    if (imgFile) {
+      const url = await uploadLivesiteLogoToStorage(imgFile);
+      setLivesiteData({
+        ...livesiteData,
+        lastEdited: getTimestamp(),
+        lastEditedBy: user,
+        imgUrl: url,
+      });
+      updateLivesiteData({
+        ...livesiteData,
+        lastEdited: getTimestamp(),
+        lastEditedBy: user,
+        imgUrl: url,
+      });
+    } else {
+      await updateLivesiteData({
+        ...livesiteData,
+        lastEdited: getTimestamp(),
+        lastEditedBy: user,
+      });
+    }
+    setisEditing(false);
+  };
+
+  const selectImageFile = (e) => {
+    if (e.target.files[0]) {
+      setImgFile(e.target.files[0]);
+      fileUpload.imgName = e.target.files[0].name;
+    }
+  };
+
+  const getAsyncData = async () => {
+    setLivesiteData(await getLivesiteData());
+  };
+
+  useEffect(() => {
+    getAsyncData();
+  }, []);
+  return (
+    <Page
+      currentPath="Livesite"
+      hackathons={hackathons}
+      navbarItems={LIVESITE_NAVBAR}
+    >
+      <Card>
+        <CardHeader>
+          <CardTitle>Livesite Settings</CardTitle>
+          <p>
+            {`Last edited by ${livesiteData.lastEditedBy} at ${formatDate(
+              livesiteData.lastEdited?.seconds
+            )}`}
+          </p>
+          <CardButtonContainer>
+            <Button type={EDIT} onClick={() => setisEditing(true)} />
+          </CardButtonContainer>
+        </CardHeader>
+        <CardContent>
+          {isEditing ? (
```

Comment:
still confused as to what the `isEditing` flag here does

Reply:
In the screenshot u can see the editing state and the non editing state